openapi-annotation-processor/src/main/kotlin/io/javalin/openapi/processor/generators/OpenApiGenerator.kt

```diff
@@ -285,14 +285,39 @@ internal class OpenApiGenerator {
         return parameter
     }
 
-    private fun generateOperationId(httpMethod: HttpMethod, openApi: OpenApi): String =
+    /**
+     * Generates a operationId for the given [HttpMethod] and [OpenApi]-route.
+     * The pattern used is `methodPathPartsCamelCaseByPathParam`.
+     *
+     * @param httpMethod The HTTP method to use. Prefixes the operationId
+     * @param openApi The route to generate the operation id for
+     * @param pathParamPrefix The prefix for path parameters to use. Defaults to `By`
+     */
+    private fun generateOperationId(
+        httpMethod: HttpMethod,
+        openApi: OpenApi,
+        pathParamPrefix: String = "By"
+    ): String =
         when (openApi.operationId) {
             AUTO_GENERATE -> {
                 httpMethod.name.lowercase() + openApi.path.split('/')
-                    .map { pathPart -> pathPart.replaceFirstChar { it.titlecase(Locale.getDefault()) } }
+                    .map { pathPart ->
+                        if (pathPart.startsWith('{') or pathPart.startsWith('<')) {
+                            /* Case this is a path parameter */
+                            val pathParam = pathPart.substring(1, pathPart.length - 1)
+                                .replaceFirstChar { it.titlecase(Locale.getDefault()) }
+                            return@map pathParamPrefix + pathParam
+                        } else {
+                            /* Case this is a regular part of the path */
+                            return@map pathPart.replaceFirstChar {
+                                it.titlecase(Locale.getDefault())
+                            }
+                        }
```

Comment:
1. `return@map` should be unnecessary, in Kotlin `if`/`when` are expressions, so the last statement in given body also returns value
2. I'd prefer `drop(1)` & `dropLast(1)` instead of `substring`

Reply:
TIL I learned about `drop` and `dropLast`. The `return@map` are propably a matter of personal style: I think it enhances readability from a certain point of nesting, hence I put them there. Nevertheless, both your points are addressed.

Reply:
I personally prefer expressions over statements, because then you're forced to provide one execution flow and there's no unexpected behavior caused by some kind of early jumps (returns/breaks/continue) that skips the rest of body:

```kotlin
val value = when {
    pathPart.startsWith('{') or pathPart.startsWith('<') ->
        pathParamPrefix + pathPart
           .drop(1)
           .dropLast(1)
           .replaceFirstChar { it.titlecase(Locale.getDefault()) }
   else ->
      pathPart.replaceFirstChar { it.titlecase(Locale.getDefault()) }
}
// [...]
```

Kinda offtopic, but it describes a little bit of the background behind my suggestion :P